main.go

```diff
@@ -222,14 +230,31 @@ func searchChart(r []*search.Result, name string, chartVersion string, devel boo
 
 			constrain, err := semver.NewConstraint(constrainStr)
 			if err != nil {
-				return nil, err
+				return nil, false, err
 			}
 
 			debug("Comparing version of original chart '%s' => %s with version (%s) %s",
 				name, chartVersion, result.Name, result.Chart.Metadata.Version)
 			debug("Using '%s' as constrain against '%s'", constrainStr, result.Chart.Metadata.Version)
 			if constrain.Check(version) {
-				return result, nil
+				// only return if 'deprecationInfo' is set to false
+				// because then we do not have to chec if there
+				// are deprecated repositories.
+				if !deprecationInfo {
+					return result, false, nil
+				} else {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [if block ends with a return statement, so drop this else and outdent its block](https://app.codacy.com/manual/l0nax/helm-whatup/pullRequest?prid=4860442)